src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -369,8 +380,18 @@ private static bool TryParseEnumCore(ref Utf8JsonReader reader, JsonSerializerOp
             int charsWritten = reader.CopyString(charBuffer);
             ReadOnlySpan<char> source = charBuffer.Slice(0, charsWritten);
 
-            // Try parsing case sensitive first
-            bool success = Enum.TryParse(source, out T result) || Enum.TryParse(source, ignoreCase: true, out result);
+            bool success;
+            T result;
+            if ((_converterOptions & EnumConverterOptions.AllowNumbers) != 0 || !NumericRegex().IsMatch(source))
```

Comment:
Shouldn't the `NumericRegex()` method be storing the result in `s_numericRegex` for consistency?

Reply:
Fixed.